corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js

```diff
@@ -496,11 +496,10 @@ hqDefine("geospatial/js/case_grouping_map",[
             // reset the warning banner
             self.groupsLocked(!self.groupsLocked());
             if (self.groupsLocked()) {
-                mapModel.mapInstance.scrollZoom.disable();
                 setCaseGroups();
             } else {
-                mapModel.mapInstance.scrollZoom.enable();
                 clearCaseGroups();
+                mapModel.fitMapBounds(mapModel.caseMapItems());
```

Comment:
What does this do exactly?

Reply:
That takes the list of map items and then creates a bounding box by starting with the first element and then looping through the rest of the array. Once the bounding box has been created, it is applied to the map which then zooms it to fit the bounds.

For more context, you can refer to the function definition [here](https://github.com/dimagi/commcare-hq/blob/ee27cab0d861829360f1c29688aee782b3da2c6a/corehq/apps/geospatial/static/geospatial/js/models.js#L482).